ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -849,8 +849,9 @@ FEDataManager::spread(const int f_data_idx,
         // Multiply by the nodal volume fractions (to convert densities into
         // values).
         NumericVector<double>& F_dX_vec = F_vec;
-        const NumericVector<double>& dX_vec = *buildDiagonalL2MassMatrix(system_name);
-        F_dX_vec.pointwise_mult(F_vec, dX_vec);
+        std::unique_ptr<NumericVector<double> > dX_vec = F_dX_vec.clone();
+        *dX_vec = *buildDiagonalL2MassMatrix(system_name);
+        F_dX_vec.pointwise_mult(F_vec, *dX_vec);
```

Comment:
The easiest way to get this to work is to ensure that the vector containing the diagonal entries of the mass matrix has exactly the same parallel layout as the other vectors involved in the operation.